x-pack/plugins/endpoint/public/embeddables/resolver/view/use_camera.ts

```diff
@@ -282,26 +282,28 @@ export function useCamera(): {
  */
 function useAutoUpdatingClientRect(): [DOMRect | null, (node: Element | null) => void] {
   const [rect, setRect] = useState<DOMRect | null>(null);
-  const nodeRef = useRef<Element | null>(null);
+  // Using state as ref.current update does not trigger effect hook when reset
+  const [currentNode, setCurrentNode] = useState<Element | null>(null);
+
   const ref = useCallback((node: Element | null) => {
-    nodeRef.current = node;
+    setCurrentNode(node);
     if (node !== null) {
       setRect(node.getBoundingClientRect());
     }
   }, []);
   const { ResizeObserver } = useContext(SideEffectContext);
   useEffect(() => {
-    if (nodeRef.current !== null) {
+    if (currentNode !== null) {
```

Comment:
nice catch. Any chance you could put a test in to validate the approach? 
This is the file that tests use_camera.
https://github.com/elastic/kibana/blob/master/x-pack/plugins/endpoint/public/embeddables/resolver/view/use_camera.test.tsx

Reply:
Yep, added.